frameworks/MLNet/exec.py

```diff
@@ -0,0 +1,117 @@
+# import standard_lib
+import tempfile
+import os
+import logging
+import psutil
+import json
+
+# import 3rd_parties
+import pandas as pd
+
+# import amlb
+from amlb.benchmark import TaskConfig
+from amlb.data import Dataset
+from amlb.results import NoResultError, save_predictions
+from amlb.utils import run_cmd, Timer
+from frameworks.shared.callee import output_subdir, save_metadata
+
+log = logging.getLogger(__name__)
+
+def run(dataset: Dataset, config: TaskConfig):
+    log.info(f"\n**** MLNet [v{config.framework_version}] ****\n")
+    save_metadata(config)
+
+    avaible_task_list = ['classification', 'regression']
+    if config.type not in avaible_task_list:
+        raise ValueError(f'{config.type} is not supported.')
+
+    dir_path = os.path.dirname(os.path.realpath(__file__))
+    DOTNET_INSTALL_DIR = os.path.join(dir_path, 'lib')
+    os.environ['DOTNET_ROOT'] = DOTNET_INSTALL_DIR
+    os.environ['MLNetCLIEnablePredict'] = 'True'
+    threads_count_per_core = psutil.cpu_count() / psutil.cpu_count(logical=False)
+    os.environ['MLNET_MAX_THREAD'] = str(config.cores * threads_count_per_core)
+    mlnet = os.path.join(DOTNET_INSTALL_DIR, 'mlnet')
+    train_time_in_seconds = config.max_runtime_seconds
+    sub_command = config.type
+
+    # set up MODELBUILDER_AUTOML
+    MODELBUILDER_AUTOML = config.framework_params.get('automl_type', 'NNI')
+    os.environ['MODELBUILDER_AUTOML'] = MODELBUILDER_AUTOML
+    is_save_artifacts = config.framework_params.get('_save_artifacts', False)
+    tmpdir = tempfile.mkdtemp()
+    if is_save_artifacts:
+        tmpdir = output_subdir('artifacts', config=config)
+
+    train_dataset_path: str
+    test_dataset_path: str
+    label: str = 'label'
+    temp_output_folder = os.path.join(tmpdir, str(config.fold))
+    log_path = os.path.join(temp_output_folder, 'log.txt')
+    label = dataset.target.index
+
+    if dataset.train.format == 'csv':
+        train_dataset_path = dataset.train.path
+        test_dataset_path = dataset.test.path
+    else:
+        # .arff
+        train_dataset_path = os.path.join(tmpdir, f'train.csv')
+        test_dataset_path = os.path.join(tmpdir, f'test.csv')
+        columns = [f'col_{i}' for i in range(dataset.train.data.shape[-1])]
+        columns[label] = 'label'
+
+        # might use a bit extra memory, but it's faster
+        pd.DataFrame(dataset.train.data, columns=columns).to_csv(train_dataset_path, index=None)
```

Comment:
After comparing between 1) directly saving ndarray into csv and 2) create df and call `df.to_csv` instead, it appears that the 2) is still a better option because it's much faster and by avoiding creating a variable to save df, the memory will be released immediately after `df.to_csv` get called. Because `dataset.train.data` is a lazy property, dataset will only be loaded into memory when get called and will only be loaded once so perf issue here should be small.